src/crimson/osd/shard_services.cc

```diff
@@ -106,12 +106,17 @@ seastar::future<> ShardServices::send_to_osd(
 
 seastar::future<> ShardServices::dispatch_context_transaction(
   crimson::os::CollectionRef col, PeeringCtx &ctx) {
-  logger().debug("ShardServices::dispatch_context_transaction: do_transaction...");
-  auto ret = store.do_transaction(
-    col,
-    std::move(ctx.transaction));
-  ctx.reset_transaction();
-  return ret;
+  if (ctx.transaction.empty()) {
+    logger().debug("ShardServices::dispatch_context_transaction: flush...");
+    return store.flush(col);
```

Comment:
Do we need to flush here if the transaction is empty?

Reply:
No, if the transaction is actually empty here I don't think we need to do anything.